File: programmers/heap/more_spicy.py
import heapq

def solution(scoville, k):
    answer = 0
    heapq.heapify(scoville)

    while 1:
        tmp1 = heapq.heappop(scoville)

        if tmp1 >= k:
            heapq.heappush(scoville, tmp1)
            break
        
        if not scoville:
            answer = -1
            break

        tmp2 = heapq.heappop(scoville)
        
        heapq.heappush(scoville, tmp1 + 2 * tmp2)
        answer += 1
    
    return answer

File: programmers/heap/more_spicy.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>();
        for (int s: scoville) {
            pq.offer(s);
        }

        while (true) {
            if (pq.size() == 0) {
                answer = -1;
                break;
            }
            
            int s1, s2;

            if (pq.peek() < K) {
                s1 = pq.poll();
            } else {
                break;
            }

            if (pq.size() == 0) {
                answer = -1;
                break;
            } else {
                s2 = pq.poll();
            }
            
            pq.offer(s1 + 2 * s2);
            answer++;
        }
        
        return answer;
    }
}